docs/translations.md

```diff
@@ -0,0 +1,103 @@
+
+Example:
+
+```yml
+en:
+  roles:
+    new:
+      invite_a_new_user: Invite a New User
+      nonprofit_admin: Nonprofit Admin
+      ...
+```
+
+3. If **there aren't** any translation files for the strings that you want to make translatable, then you need to create a new source file. The name of the translation file should be the same as the component name, but with `.en` in the and, followed by the `.yml` extension. For example, [app/views/roles/_new_modal.html.erb](app/views/roles/_new_modal.html.erb)'s translation file is named as `roles.new.en.yml`.
+
+Example:
+
+```yml
+en:
+  roles:
+    new:
+      invite_a_new_user: Invite a New User
+      nonprofit_admin: Nonprofit Admin
+      ...
+```
+
+4. Add the new file(s) you created to the [Crowdin config file](../crowdin.yml) indicating the path for the source language and the path for the translations.
+
+Example:
+
+```yml
+files:
+  - source: /**/locales/en.yml
+    translation: /**/locales/%locale%.yml
+  - source: /**/locales/roles.new.en.yml
+    translation: /**/locales/roles.new.%locale%.yml
+```
+
+5. Find every string that is going to be translated, create a key for it and place it into your translation file.
+
+It's important to give it a well detailed name, so that when someone is reading the code on the component they automatically understand what that string is without having to look at the translation file.
+
+For example, the text `Full permissions, including bank payouts` on [roles.new.yml](config/locales/roles.new.en.yml), as big as it is, has the key `full_permissions_including_bank_payouts`.
+
+6. Replace the strings from the components with their corresponding keys.
+
+Examples:
+
+* On .erb files, you should replace the "Full permissions, including bank payouts" text for `t('roles.new.full_permissions_including_bank_payouts')`;
+* For `.tsx` files, you should import the `useIntl` hook from the `intl` component and import `formatMessage` from it. You should then create a constant with `formatMessage({ id: 'roles.new.full_permissions_including_bank_payouts' })` and replace the text by the name of the constant you created - you can look at how it's done on our [SignInComponent.tsx](https://github.com/houdiniproject/houdini/blob/a31e755f1bdf21c5c894018fe8ec3b26fcf6c896/app/javascript/components/users/SignInComponent.tsx#L77);
+* For `.js` or `.es6` files, replace the text with `I18n.t('roles.new.full_permissions_including_bank_payouts')`.
+
+7. Commit the changes and open a pull request to Houdini/main.
+
+8. Once the pull request is merged, after up to 1 hour, Crowdin is going to sync the new file and make it available for translations. Once the file is synced, you can start translating from Crowdin.
```

Comment:
I considered as we would be using our current file organization instead of having folders for each language. I saw some other Rails projects organize like that, so I think we could keep it. [Discourse](https://github.com/discourse/discourse/tree/main/config/locales) is an example, and they use their Crowdin instance for that.

Here's the text if we change our minds and decide to change the organization:

```suggestion
We can also have translations for strings that are shared in multiple components. For example, [config/locales/ui.en.yml](config/locales/en-EN/ui.yml) holds UI strings that can be shared between multiple components, such as common button labels or links.

2. If **there are** translation files where the strings you want to make translatable can be placed (given the context and the organization of the files), you can add the keys and original text to this file (see the example below). **After adding the new keys and strings, you can go to step 4.**

Example:

```yml
en:
  roles:
    new:
      invite_a_new_user: Invite a New User
      nonprofit_admin: Nonprofit Admin
      ...
\```

3. If **there aren't** any translation files for the strings that you want to make translatable, then you need to create a new source file. The name of the translation file should be the same as the component name, but with `.en` in the and, followed by the `.yml` extension. For example, [app/views/roles/_new_modal.html.erb](app/views/roles/_new_modal.html.erb)'s translation file is named as `roles.new.en.yml`.

Example:

```yml
en:
  roles:
    new:
      invite_a_new_user: Invite a New User
      nonprofit_admin: Nonprofit Admin
      ...
\```

4. Find every string that is going to be translated, create a key for it and place it into your translation file.

It's important to give it a well detailed name, so that when someone is reading the code on the component they automatically understand what that string is without having to look at the translation file.

For example, the text `Full permissions, including bank payouts` on [roles.new.yml](config/locales/en-EN/roles.new.yml), as big as it is, has the key `full_permissions_including_bank_payouts`.

5. Replace the strings from the components with their corresponding keys.

Examples:

* On .erb files, you should replace the "Full permissions, including bank payouts" text for `t('roles.new.full_permissions_including_bank_payouts')`;
* For `.tsx` files, you should import the `useIntl` hook from the `intl` component and import `formatMessage` from it. You should then create a constant with `formatMessage({ id: 'roles.new.full_permissions_including_bank_payouts' })` and replace the text by the name of the constant you created - you can look at how it's done on our [SignInComponent.tsx](https://github.com/houdiniproject/houdini/blob/a31e755f1bdf21c5c894018fe8ec3b26fcf6c896/app/javascript/components/users/SignInComponent.tsx#L77);
* For `.js` or `.es6` files, replace the text with `I18n.t('roles.new.full_permissions_including_bank_payouts')`.

6. Commit the changes and open a pull request to Houdini/main.

7. Once the pull request is merged, after up to 1 hour, Crowdin is going to sync the new file and make it available for translations. Once the file is synced, you can start translating from Crowdin.
```